storage/aptosdb/src/backup/restore_handler.rs

```diff
@@ -132,7 +132,12 @@ impl RestoreHandler {
     }
 
     pub fn get_in_progress_state_kv_snapshot_version(&self) -> Result<Option<Version>> {
-        let db = self.aptosdb.ledger_db.metadata_db_arc();
+        let sharding_enabled = self.aptosdb.ledger_db.enable_storage_sharding();
+        let db = if !sharding_enabled {
+            self.aptosdb.ledger_db.metadata_db_arc()
+        } else {
+            self.aptosdb.state_kv_db.metadata_db_arc()
```

Comment:
looks like metadata_db_arc() will return the correct db no matter if sharding is enabled?